lib/cchain/provider/cache.go

```diff
@@ -0,0 +1,86 @@
+package provider
+
+import (
+	"sync"
+
+	"github.com/omni-network/omni/lib/xchain"
+)
+
+// chainVersionCache contains approved attestations of a chain version with all heights above the watermark.
+type chainVersionCache struct {
+	cache     map[uint64]xchain.Attestation
+	watermark uint64
+}
+
+// attestationCache if a thread-safe collection of caches containing attestations per chain version.
+// All cache sizes will be kept under `cacheSize` elements.
+type attestationCache struct {
+	cacheSizes int
+	caches     map[xchain.ChainVersion]*chainVersionCache
```

Comment:
i'd personally use a slice for performance reasons, but maybe that is overkill (and adds complexity), might be worth doing a benchmark teststo compare slices vs maps